### PR TITLE
PUPIL-573 feat(unitlistingquery): create unit listing query

### PR DESCRIPTION
**Pull request by benprotheroe**
## Description

Music year: 1987

create unitlistingQuery 

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2385--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Screenshots

How it used to look (delete if n/a):
{screenshots}

How it should now look:
{screenshots}

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.schema.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { z } from "zod";
+import { syntheticUnitvariantsWithLessonIdsSchema } from "@oaknational/oak-curriculum-schema";
+
+import { ConvertKeysToCamelCase } from "@/utils/snakeCaseConverter";
+
+export const unitBrowseDataSchema = z.array(
+  syntheticUnitvariantsWithLessonIdsSchema.omit({
+    null_unitvariant: true,
```
Comment (kimon-satan) [inline src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.schema.ts:8]: We might actually need the null_unitvariant when it comes to implementing optionality. I can't remember so this is fine for now.

File: src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  unitDataFixture,
+  unitvariantFixture,
+} from "@oaknational/oak-curriculum-schema";
+
+import { pupilUnitListingQuery } from "./pupilUnitListing.query";
+
+import { PupilUnitListingQuery } from "@/node-lib/curriculum-api-2023/generated/sdk";
+import sdk from "@/node-lib/curriculum-api-2023/sdk";
+
+export const syntheticUnitvariantsWithLessonIdsFixture = ({
```
Comment (kimon-satan) [inline src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.test.ts:13]: I'm confused why this is here. Didn't you just do an update to oak-curriculum-schema to export this from there ?

File: package.json [hunk 3]
```diff
@@ -70,7 +70,7 @@
     "@mux/mux-node": "^7.3.3",
     "@mux/mux-player-react": "^2.3.1",
     "@oaknational/oak-components": "^0.16.1",
-    "@oaknational/oak-curriculum-schema": "^1.2.0",
+    "@oaknational/oak-curriculum-schema": "^1.3.0",
```
Comment (kimon-satan) [inline package.json:73]: I think you need to incorporate your latest changes to oak-curriculum-schema

**Review by kimon-satan — CHANGES_REQUESTED** [review]

**Comment by RBrownsett** [comment]
@benprotheroe should the new query be returning the correct unit order for programmes that have an exam board and differentiate by tier? Only I ran the below query for chemistry higher edexcel and the unit order did not match what was on the front end?

query MyQuery {
  published_mv_synthetic_unitvariants_with_lesson_ids_by_year_6_0_0(where: {programme_slug: {_eq: "chemistry-secondary-year-10-higher-edexcel"}}) {
    lesson_count
    unit_data(path: "title")
    unit_slug
    supplementary_data(path: "unit_order")
    programme_slug
  }
}

Results (unit order 4):

image

Front End (Unit order 2)

image

File: src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.test.ts (hunk 2 above)
Comment (benprotheroe) [reply src/node-lib/curriculum-api-2023/queries/pupilUnitListing/pupilUnitListing.test.ts:13]: ill update now

**Review by RBrownsett — APPROVED** [review]
Tested in legacy running the query to check:
-Units are now split out by year when provided with a year based programme slug
-Number of lessons, unit slug and title is shown correctly

Note: Ordering will be done via the front end for when a unit has lessons that have not yet been published (this answers the question of my previous comment where the unit order did not match due to lessons not yet published)

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**